saleor/graphql/notifications/tests/test_external_notification_query.py

```diff
@@ -0,0 +1,156 @@
+import json
+from unittest.mock import patch
+
+import pytest
+from graphql_relay.node.node import to_global_id
+
+from ....account.models import User
+from ....core.notify_events import UserNotifyEvent
+from ....graphql.tests.utils import assert_no_permission
+from ....plugins.tests.sample_plugins import PluginSample
+
+query_test_data = [
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            },
+            "pluginId": "",
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            },
+            "pluginId": "WRONG-TEST-PLUGIN",
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            }
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+            }
+        },
+        400,
+    ),
+    (
+        {
+            "input": {
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            }
+        },
+        400,
+    ),
+    ({"input": {"ids": [], "externalEventType": {}, "channel": "c-pln"}}, 200),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "channel": "c-pln",
+            }
+        },
+        400,
+    ),
+]
+
+
+@pytest.mark.parametrize("variables, status_code", query_test_data)
+def test_query(
+    variables,
+    status_code,
+    external_notification_trigger_query,
+    staff_api_client,
+    permission_manage_users,
+    channel_PLN,
+):
+    response = staff_api_client.post_graphql(
+        external_notification_trigger_query,
+        variables,
+        permissions=[permission_manage_users],
+        check_no_permissions=False,
+    )
+    assert response.status_code == status_code
+
+
+@patch("saleor.plugins.manager.PluginsManager.notify")
+def test_notify_via_external_notification_trigger_for_all_plugins(
```

Comment:
I think the name is not correct. It doesn't test mutation for all plugins. It just test the mutation and confirms that pluginmanager is called.